### PR TITLE
Add typescript definition and Pug support

### PR DESCRIPTION
**Pull request by patarapolw**
Also, correct `README.md` on the error

https://github.com/highlightjs/highlightjs-vue/issues/1#issue-523817836

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +23,10 @@
   },
   "homepage": "https://github.com/highlightjs/highlightjs-vue#readme",
   "dependencies": {
-    "highlight.js": "^9.15.10",
+    "@types/highlight.js": "^9.12.3",
```
Comment (marcoscaceres) [inline package.json]: I don't know typescript, but wondering if this should also be a dev dependency?

Comment (patarapolw) [reply package.json]: Not sure about this, but I believe it should be `"dependencies"`, as users don't have to install `@types/highlight.js` themselves, when needing types.

Comment (LissetteIbnz) [reply package.json]: @types its only for TypeScript types and must be dev dependency. 
highlight.js is necessary be installed also as dependency.

File: vue.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*
 Language: Vue.js
-Requires: xml.js, javascript.js, typescript.js, css.js, stylus.js, scss.js
+Requires: xml.js, pug.js, javascript.js, typescript.js, css.js, stylus.js, scss.js
```
Comment (LissetteIbnz) [inline vue.js]: highlight.js not support pug.

**Comment by LissetteIbnz** [comment]
It turns out that Jade/Pug is not supported in highlight.js. I'll close this PR.

@patarapolw, if you feel this support is necessary, highlight.js might be a better place for a feature request.

https://github.com/highlightjs/highlight.js/issues/1159

File: package.json (hunk 1 above)
Comment (patarapolw) [reply package.json]: Thanks. It seems so. This is relevant and up to date -- https://stackoverflow.com/questions/45176661/how-do-i-decide-whether-types-goes-into-dependencies-or-devdependencies

**Comment by patarapolw** [comment]
> It turns out that Jade/Pug is not supported in highlight.js. I'll close this PR.

I see, it seems to be [a failed attempt](https://github.com/highlightjs/highlight.js/issues/1159#issuecomment-214911344), and no one tried anymore.

**Pull request closed without merging**